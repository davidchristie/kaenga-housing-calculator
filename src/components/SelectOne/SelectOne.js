import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Button } from 'react-bootstrap'

class SelectOne extends Component {
  render () {
    const {
      input: { onChange, value },
      options
    } = this.props
    return (
      <div>
        {options.map((option, index) => {
          return (
            <Button
              active={value === option.value}
              key={index}
              onClick={() => onChange(option.value)}
            >
              {option.text}
            </Button>
          )
        })}
      </div>
    )
  }
}

SelectOne.defaultProps = {
  input: {},
  options: []
}

SelectOne.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired
}

export default SelectOne
