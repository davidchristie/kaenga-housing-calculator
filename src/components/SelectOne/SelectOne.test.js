import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import SelectOne from './SelectOne'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<SelectOne />)).toMatchSnapshot()
})
