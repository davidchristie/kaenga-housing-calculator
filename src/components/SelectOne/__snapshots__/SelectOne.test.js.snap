exports[`test matches snapshot 1`] = `
<div>
  
</div>
`;
